packages/eslint-plugin/docs/rules/no-floating-promises.mdx

```diff
@@ -84,6 +84,53 @@ await Promise.all([1, 2, 3].map(async x => x + 1));
 
 ## Options
 
+### `checkThenables`
```

Comment:
(optional - up to your judgement)
I personally would love to see it spelled out that there are really two cases.
1. Traditional Thenables which are intended to act like promises, and therefore _should_ be subject to the same checks around rejection that ordinary Promises are. (which is probably not terribly important anymore, since people are probably almost all using ordinary Promises, or at least ordinary Promise _types_ with a polyfill provided by a build step, but probably _not_ really using custom Promise polyfill types anymore very frequently) 
1. Custom objects that (ab)use the Thenable/async/await protocol to hijack the `await` syntax for behavior which may bear no resemblance at all to Promises. (of which the only example I've heard come up in discussions is the fastify case)

Reply:
This is good info! Linking to #8517: I think I'll want to put deeper dives / contextual info in the dedicated docs.